src/main/java/life/qbic/model/download/QbicDataDownloader.java

```diff
@@ -249,6 +269,14 @@ private void downloadFile(DataSetFile dataSetFile, Path prefix) throws DownloadE
       } else {
         LOG.info("Skipped empty file " + file.getDataSetFile().getPath());
       }
+
+      // consume the rest of the stream
+      while (Objects.nonNull(file = reader.read())) {
+        // we only expected exactly one file
+        LOG.debug("Ignored file " + file.getDataSetFile().getPath() + "; Expected one file exactly");
+      }
+    } finally {
+      reader.close();
```

Comment:
why is this necessary?

Reply:
as soon as the stream is consumed, stuff happens in the openbis reader.
This makes sure that nothing is missed. Plus we get to log unexpected behavior.
Should never happen but if it happens we don't fail but report it into the log (not to the console)